content/en/blog/_posts/2021-04-21-Graceful-Node-Shutdown-Beta.md

```diff
@@ -0,0 +1,66 @@
+
+## How does it work?
+On Linux, your system can shut down in many different situations. For example:
+* A user or script running `shutdown -h -P now` or `systemctl poweroff`
+* Physically pressing a power button on the machine.
+* Stopping a VM instance on a cloud provider, e.g. `gcloud compute instances stop` on GCP.
+* A Preemptible VM or Spot Instances that can be terminated by a cloud provider unexpectedly.
+
+Many of these situations can be unexpected and there is no guarantee that a cluster administrator drained the node prior to these events. With the  Graceful node shutdown feature, kubelet uses a systemd mechanism called "[Inhibitor Locks](https://www.freedesktop.org/wiki/Software/systemd/inhibit)" to allow draining in most cases. Using Inhibitor Locks, kubelet instructs systemd to postpone system shutdown for a specified duration, giving a chance for the node to drain and evict pods on the system.
+
+Kubelet makes use of this mechanism to ensure your pods will be terminated cleanly. When the kubelet starts, it acquires a systemd delay-type inhibitor lock. When the system is about to shut down, the kubelet can delay that shutdown for a configurable, short duration utilizing the delay-type inhibitor lock it acquired earlier. This gives your pods extra time to  terminate. As a result, even during unexpected shutdowns, your application will receive a SIGTERM, preStop hooks will execute, and kubelet will properly update it’s own ready status and pod statuses to the api-server.
+
+For example, on a node with Graceful node shutdown enabled, you can see that the inhibitor lock is actively taken by the kubelet:
+
+```
+kubelet-node ~ # systemd-inhibit --list
+    Who: kubelet (UID 0/root, PID 1515/kubelet)
+    What: shutdown
+    Why: Kubelet needs time to handle node shutdown
+    Mode: delay
+
+1 inhibitors listed.
+```
+
+One important consideration we took when designing this feature is that not all pods are created equal. For example, some of the pods running on a node such as a logging related daemonset should stay running as long as possible to capture important logs during the shutdown itself. As a result, pods are split into two categories: “regular” and “critical”. [Critical pods](https://kubernetes.io/docs/tasks/administer-cluster/guaranteed-scheduling-critical-addon-pods/#marking-pod-as-critical) are those that have `priorityClassName` set to `system-cluster-critical` or `system-node-critical`; all other pods are considered regular. In our example, the logging DaemonSet would run as a critical pod. During the graceful shutdown, regular pods are terminated first, followed by critical pods. As an example, this would allow a critical pod like those part of a logging daemonset to continue functioning, and collecting logs during the termination of regular pods.
```

Comment:
```suggestion
One important consideration we took when designing this feature is that not all pods are created equal. For example, some of the pods running on a node such as a logging related daemonset should stay running as long as possible to capture important logs during the shutdown itself. As a result, pods are split into two categories: “regular” and “critical”. [Critical pods](/docs/tasks/administer-cluster/guaranteed-scheduling-critical-addon-pods/#marking-pod-as-critical) are those that have `priorityClassName` set to `system-cluster-critical` or `system-node-critical`; all other pods are considered regular.

In our example, the logging DaemonSet would run as a critical pod. During the graceful shutdown, regular pods are terminated first, followed by critical pods. As an example, this would allow a critical pod like those part of a logging DaemonSet to continue functioning, and collect logs throughout the termination of regular pods.
```

Reply:
Done, line break is helpful here.